metabase/metabase
# Dataset query editor

Once a saved question is turned into a dataset, there is no way to edit the underlying query. It's intentional, so datasets behave as raw tables opened in simple mode (see #18993 and #19101).

This PR introduces a dataset editor page available on `/dataset/:id-:slug/query`, where people can edit their dataset queries

### To Verify

For GUI and native questions:

1. Turn a question into a dataset (following the steps from #18702)
2. Open the dataset details sidebar (click the dataset name while being on the `/dataset/:id-:slug` page
3. Click "Edit query definition"
4. Make sure you are navigated to a new page and it looks as expected
    * the global navbar is not shown
    * there is a query editor (notebook for structured queries and native editor for native queries)
    * there is a results table
    * you can resize the editor using a handle
5. Edit and re-run the query and click "Cancel" in the top right
    * you should be navigated back to `/dataset/:id-:slug`
    * the results table should have the results for the original query
    * open the editor again, you should  see the original query without any changes
6. Edit the query and click "Save changes"
    * you should be navigated back to `/dataset/:id-:slug`
    * the results table should now have the results for the updated query
    * open the editor again, you should  see the updated query

### Demo

**Structured Query**

https://user-images.githubusercontent.com/17258145/143591558-91f79ed9-59d1-4c23-a6ed-f6a57962c072.mov

**Native Query**

https://user-images.githubusercontent.com/17258145/143591571-895b2c95-f9de-4e00-9bbd-fdbb829c08cc.mov
